WooCommerce/Classes/Yosemite/DefaultStoresManager.swift

```diff
@@ -50,6 +51,11 @@ class DefaultStoresManager: StoresManager {
         return state is AuthenticatedState
     }
 
+    var isLoggedIn: Observable<Bool> {
+        isLoggedInSubject
+    }
+    private let isLoggedInSubject = BehaviorSubject<Bool>(false)
+
```

Comment:
yea I'm excited to see more observable states in the codebase!

> Observing if the current siteID changed. Effect: Delete all the controllers!

yes! currently, the site info is in `SessionManager` and I plan to add an observable there.

> Observing if a specific order changes like storesManager.orders[orderID].subscribe { ... }
> Create ResultsController objects like storesManager.orders.resultsControllerForOrderList(). Effect: We no longer have a reference to Storage in WooCommerce! Of course, this assumes that DefaultStoresManager is part of Yosemite, which it isn't. 🙈 Still a lot to work on here!
> Read-only queries without ResultsController like storesManager.orders.orderWith(id:)

I think FRC and `EntityListener` are really the only reasons we have to expose Storage / Core Data in the app layer. It'd be really cool if we could have a way to observe the storage layer changes in Yosemite to replace FRC and `EntityListener`. `FetchResultSnapshotsProvider` feels like a great start with an observable snapshot!

One tricky part to replace FRC I can think of is to translate a query in terms of readonly models to storage layer predicate. Like from a predicate `Yosemite.Product.productType == .simple` to `Storage.Product.productTypeKey == "simple"` since some fields don't match exactly due to Core Data model constraints. We might have to create a function for each use case like `storesManager.products.productsWith(productType:)`

Reply:
Creating `UseCase`s to encapsulate the query sounds cool, I'm all for it for Yosemite to have most of the business logic and App layer to just handle UI! 👍